src/modules/Sticky/Sticky.js

```diff
@@ -0,0 +1,245 @@
+import _ from 'lodash'
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+
+import {
+  customPropTypes,
+  getElementType,
+  getUnhandledProps,
+  META,
+} from '../../lib'
+
+/**
+ * Sticky content stays fixed to the browser viewport while another column of content is visible on the page.
+ */
+export default class Sticky extends Component {
+  static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Offset in pixels from the bottom of the screen when fixing element to viewport. */
+    bottomOffset: PropTypes.number,
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Context which sticky element should stick to. */
+    context: PropTypes.object,
+
+    /** Offset in pixels from the top of the screen when fixing element to viewport. */
+    offset: PropTypes.number,
+
+    /**
+     * Callback when element is bound to bottom of parent container.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onBottom: PropTypes.func,
+
+    /**
+     * Callback when element is fixed to page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onStick: PropTypes.func,
+
+    /**
+     * Callback when element is bound to top of parent container.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onTop: PropTypes.func,
+
+    /**
+     * Callback when element is unfixed from page.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All props.
+     */
+    onUnstick: PropTypes.func,
+
+    /** Whether element should be "pushed" by the viewport, attaching to the bottom of the screen when scrolling up. */
+    pushing: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    bottomOffset: 0,
+    offset: 0,
+  }
+
+  static _meta = {
+    name: 'Sticky',
+    type: META.TYPES.MODULE,
+  }
+
+  state = {
+    sticky: false,
+  }
+
+  componentDidMount() {
+    this.handleUpdate()
+    window.addEventListener('scroll', this.handleUpdate)
```

Comment:
We've got a perf leak here that we should handle.  When scrolling the page on my macbook, this method is fired hundreds of times.  This will especially be a concern since `handleUpdate` then calls `getBoundingClientRect`.  This method [forces a reflow](https://gist.github.com/paulirish/5d52fb081b3570c81e3a) and a repaint which is a pretty costly performance hit.

One solution might be to use `requestAnimationFrame` instead.  There is a lot of info on this around the web, but [here](https://www.html5rocks.com/en/tutorials/speed/animations/) is an older article on the topic that gives one explanation.